homeassistant/components/hvac/zwave.py

```diff
@@ -0,0 +1,244 @@
+REMOTEC_ZXT_120_THERMOSTAT = (REMOTEC, REMOTEC_ZXT_120, 0)
+
+WORKAROUND_ZXT_120 = 'zxt_120'
+
+DEVICE_MAPPINGS = {
+    REMOTEC_ZXT_120_THERMOSTAT: WORKAROUND_ZXT_120
+}
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the ZWave Hvac devices."""
+    if discovery_info is None or zwave.NETWORK is None:
+        _LOGGER.debug('No discovery_info=%s or no NETWORK=%s',
+                      discovery_info, zwave.NETWORK)
+        return
+
+    node = zwave.NETWORK.nodes[discovery_info[ATTR_NODE_ID]]
+    value = node.values[discovery_info[ATTR_VALUE_ID]]
+    value.set_change_verified(False)
+    add_devices([ZWaveHvac(value)])
+    _LOGGER.debug('add_devices %s', ZWaveHvac(value))
+    _LOGGER.debug('discovery_info=%s and zwave.NETWORK=%s',
+                  discovery_info, zwave.NETWORK)
+
+
+# pylint: disable=too-many-arguments
+class ZWaveHvac(ZWaveDeviceEntity, HvacDevice):
+    """Represents a HeatControl hvac."""
+
+    # pylint: disable=too-many-public-methods, too-many-instance-attributes
+    def __init__(self, value):
+        """Initialize the zwave hvac."""
+        from openzwave.network import ZWaveNetwork
+        from pydispatch import dispatcher
+        ZWaveDeviceEntity.__init__(self, value, DOMAIN)
+        self._node = value.node
+        self._target_temperature = None
+        self._current_temperature = None
+        self._current_operation = None
+        self._operation_list = None
+        self._current_operation_state = None
+        self._current_fan_mode = None
+        self._fan_list = None
+        self._current_swing_mode = None
+        self._swing_list = None
+        self._unit = None
+        self._zxt_120 = None
+        self.update_properties()
+        # register listener
+        dispatcher.connect(
+            self.value_changed, ZWaveNetwork.SIGNAL_VALUE_CHANGED)
+        # Make sure that we have values for the key before converting to int
+        if (value.node.manufacturer_id.strip() and
+                value.node.product_id.strip()):
+            specific_sensor_key = (int(value.node.manufacturer_id, 16),
+                                   int(value.node.product_id, 16),
+                                   value.index)
+
+            if specific_sensor_key in DEVICE_MAPPINGS:
+                if DEVICE_MAPPINGS[specific_sensor_key] == WORKAROUND_ZXT_120:
+                    _LOGGER.debug('Found Remotec ZXT-120 Zwave Thermostat')
+                    self._zxt_120 = 1
+
+    def value_changed(self, value):
+        """Called when a value has changed on the network."""
+        if self._value.node == value.node:
+            self.update_properties()
+            self.update_ha_state()
+            _LOGGER.debug('Value changed on network %s)', value)
+
+    def update_properties(self):
+        """Callback on data change for the registered node/value pair."""
+        # Set point
+        for _, value in self._node.get_values(class_id=0x43).items():
```

Comment:
I think it could be possible to simplify this by using a dict which maps class_id and the corresponding property. Then iterate over that and inside that iterate over the node values. You can have one dict for single values and another dict for values in lists. Maybe something like this:
```python
class_id_property_map = {
    0x43: self._target_temperature,
    0x40: self._current_operation,
    0x31: self._current_temperature,
    ...
}
class_id_property_list_map = {
    0x40: self._operation_list,
    0x44: self._fan_list,
    ...
}
for class_id, prop in class_id_property_map.items():
    for value in self._node.get_values(class_id=class_id).values():
        if class_id == 0x43 and int(value.data) == 0:
            continue
        if class_id == 0x43 or class_id == 0x31:
            prop = int(value.data)
        if class_id == 0x31:
            self._unit = value.units
        ...
        else:
            prop = value.data

for class_id, prop in class_id_property_map.items():
    for value in self._node.get_values(class_id=class_id).values():
        if class_id == 0x44 and self._zxt_120 != 1:
            continue
        prop = list(value.data_items)
        ...
```
It might even work to call iterate over both dicts in a similar fashion, to have even less duplicate code.